rkt/rkt.go

```diff
@@ -259,28 +259,42 @@ func getDataDir() string {
 }
 
 func calculateDataDir() string {
+	var dataDir string
+
 	// If --dir parameter is passed, then use this value.
 	if dirFlag := cmdRkt.PersistentFlags().Lookup("dir"); dirFlag != nil {
 		if dirFlag.Changed {
-			return globalFlags.Dir
+			dataDir = globalFlags.Dir
 		}
 	} else {
 		// should not happen
 		panic(`"--dir" flag not found`)
 	}
 
 	// If above fails, then try to get the value from configuration.
-	if config, err := getConfig(); err != nil {
-		stderr.PrintE("cannot get configuration", err)
-		os.Exit(1)
-	} else {
-		if config.Paths.DataDir != "" {
-			return config.Paths.DataDir
+	if dataDir == "" {
+		if config, err := getConfig(); err != nil {
+			stderr.PrintE("cannot get configuration", err)
+			os.Exit(1)
+		} else {
+			if config.Paths.DataDir != "" {
+				dataDir = config.Paths.DataDir
+			}
 		}
```

Comment:
@s-urbaniak Agree. I just kept the previous logic to bring less pollution to the patch. But I'll change it if you're ok.